File: main/05_filtering_data.py
## Filtering based on values in a column
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
df = df[df['Age'] > 20]

df = df[df.Age > 20]

namecheck = ['Alfred', 'Batman']
df = df[df.Name.isin(namecheck)] # only keep the ones you want
df = df[~df.Name.isin(namecheck)] # reverse of the above


## Drop specific columns
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
df = df.drop(columns=['Age']) # drops the 'Age' column


## Drop specific rows
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
df = df.drop(index=0) # drops the first row


## Bonus: filter for NAs
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
df.at[0, 'Name'] = float('NaN') # set NA
df = df[df.Name.isnull()] # only NAs
df = df[df.name.notnull()] # only non-NAs

File: main/05_filtering_data.R
## Filtering based on values in a column
# base and/or tidyverse
library(tidyverse)
df = read_tsv('00_R_data_set.tsv')
df = df %>% filter(Age > 20)

df = subset(df, Age > 20)

# data.table
library(data.table)
dt = fread('00_R_data_set.tsv')
dt = dt[Age > 20]


## Drop specific columns
# base and/or tidyverse
library(tidyverse)
df = read_tsv('00_R_data_set.tsv')
df = df %>% select(-Age)

df$Age = NULL

# data.table
library(data.table)
dt = fread('00_R_data_set.tsv')
dt = dt[, Age:=NULL]


## Drop specific rows
# base and/or tidyverse
library(tidyverse)
df = read_tsv('00_R_data_set.tsv')
df[-1, ] # drops the first row


## Bonus: filter for NAs
# base and/or tidyverse
library(tidyverse)
df = read_tsv('00_R_data_set.tsv')
df[1, 1] = NA # set NA
df %>% filter(is.na(Name)) # only NAs
df %>% filter(!is.na(Name)) # only non-NAs

# data.table
library(data.table)
dt = fread('00_R_data_set.tsv')
dt[1, 1] = NA # set NA
dt[complete.cases(dt)]
